protocol/messenger_handler.go

```diff
@@ -262,7 +262,27 @@ func (m *Messenger) PendingNotificationContactRequest(contactID string) (*Activi
 	return m.persistence.ActiveContactRequestNotification(contactID)
 }
 
-func (m *Messenger) createContactRequestNotification(contact *Contact, messageState *ReceivedMessageState, contactRequest *common.Message) error {
+func (m *Messenger) createIncomingContactRequestNotification(contact *Contact, messageState *ReceivedMessageState, contactRequest *common.Message) error {
+	if contactRequest != nil && contactRequest.ContactRequestState == common.ContactRequestStateAccepted {
+		// Pull one from the db if there
+		notification, err := m.persistence.GetActivityCenterNotificationByID(types.FromHex(contactRequest.ID))
+		if err != nil {
+			return err
+		}
+
+		if notification != nil {
+			notification.Message = contactRequest
+			notification.Read = false
+			notification.Accepted = true
```

Comment:
According new AC logic it should always sync contactRequest  and notification state
```
notification.Accepted = contactRequest.ContactRequestState == common.ContactRequestStateAccepted
notification.Dismissed = contactRequest.ContactRequestState == common.ContactRequestStateDismissed
```